### PR TITLE
I refactored the late `MapViewModel`:

### PR DESCRIPTION
**Pull request by YofiY**
- Renamed the  `MapViewModel` to `ItineraryViewModel`
	- Created a specific View Model for itinerary creation called `CreateItineraryViewModel`

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt
Comment (ThomasPFMuller) [inline app/src/main/java/com/github/wanderwise_inc/app/viewmodel/CreateItineraryViewModel.kt:1]: I like the idea of separating this new viewmodel to improve specialization, but does it not increase complexity significantly? In my head having a viewmodel responsible for all interactions with itineraries is enough and we do not need one specialized in creating itineraries.

**Review by ThomasPFMuller — COMMENTED** [review]
Okay, so if I understand correctly, we are renaming MapViewModel to ItineraryViewModel and creating a new ViewModel to take care of the itinerary creation. Could you explain why this specialization is needed?

**Review by gunnm0 — APPROVED** [review]
I think the name change from mapViewModel to ItineraryViewModel makes a lot of sense since the class manages only itineraries. I also like the split into two classes, ItineraryViewModel and CreateItineraryViewModel makes the structure simpler to understand., having one class do everything is confusing.
I also like the fact that CreateItineraryViewModel inherits from ItineraryViewModel, this will make the code a lot simpler to write for the previewItierary of the Create screen.

**Comment by engjellismaili** [comment]
@YofiY have you done some tests for the new view model you added ? If not, it would be great to add them before merging to main

**Review by ethangraham2001 — APPROVED** [review]
Looks good. Still disagree with needing polymorphism for this as it increases complexity (more so than having five extra function definitions in the old `MapViewModel`). But as long as it works it's cool with me

**Pull request merged**